### PR TITLE
Added /wall

### PR DESCRIPTION
**Pull request by yanneves**
According to [The Big O of Code Reviews](https://www.egorand.dev/the-big-o-of-code-reviews/), this is a O(n!) change. 🙃

Adds a new page `/wall` as a foundation to automate slides to display on the day, inspired by [confwall](https://confwall.com/). 

Currently just displays the title of the soonest event:

image

But in doing so, I resurrected an effort begun 7 years ago to eradicate our dependency on the Gulp ecosystem. It wasn't clear in that generation of web tooling, but the lack of dependency discipline in packages caused an ongoing maintenance headache.

## Architecture

One aim in the architecture of this repo has been to remain true to its accessible origins as a Jekyll static site. That's maintained with these changes, plus the tools I've migrated to have a modern discipline that allow us to more easily adopt future updates and runtimes.

It's tempting to take this stepping stone to refactor the site as a modern SSG-enabled framework (Next.js, SvelteKit, Astro). I'd welcome the effort. Just bear in mind, I almost decided to migrate to Gatsby when I started this effort, and I know I'd [regret that today](https://www.smashingmagazine.com/2024/03/end-of-gatsby-journey/).

## Tailwind

Another feature of these changes is the inclusion of Tailwind support. Testing this involves a site-wide visual diff - I actually caught a lot of existing glitches in the process.

It's an opinionated change, but even being familiar with the site's code I find the styling to be a bit of a black box. Tailwind will allow us to transition to declarative styles that will be easier to maintain as we introduce other features and tweak the design.

## News

A final touch at the end here, made any active headlines pop. 📢 

firefox_xQ02hVX1Wz

## Expanded commit details

feat: Improved news styling

refactor: Added Tailwind

refactor: Dropped Gulp

Replaces several gulp tasks with more stable npm scripts

Updates metalsmith and related libraries with breaking changes

Replaces sync-files package with cpr

Moves to postcss-cli

Removes gulpsmith in preference of using metalsmith directly

Removes timed news feature rendered obsolete by metalsmith

Switches to esbuild for scripts

Switches to chokidar-cli for local dev environment

Ditches sass syntax for modern css

Upgrades to @metalsmith/collections, @metalsmith/excerpts, @metalsmith/drafts

Renders metalsmith-collections-filter, metalsmith-collections-limit packages obsolete

Swaps previous / next event props now they're corrected in @metalsmith/collections

refactor: Recovered event listings with url collisions

Split up the following pages that previously used the same slug:
- `/web-maps` `/7-similarities-between-go-and-node`
- `/react-native` `/react-native-2`
- `/webgl` `/webgl-threejs`

Previously the older event was overwritten, so this recovers some pages to appear in the Async archives once again

refactor: Removed obsolete Lanyrd widget

refactor: Removed obsolete email obfuscation script

refactor: Removed obsolete data-[slides|video] detection

This feature would retrieve an embed url for slides or videos using an unsupported (jekyll-flavoured?) markdown link syntax. It was only used for two talks' slides and the linked videos (from Vimeo) all 404.

chore: Updated dependencies

### CODE REVIEW

File: _layouts/archive.hbs [hunk 1]
```diff
@@ -1,8 +1,8 @@
 <!DOCTYPE html>
-<html>
+<html class="flex flex-1 min-h-full">
```
Comment (yanneves) [inline _layouts/archive.hbs:2]: I'm pretty sure we can drop the `flex-1` on these

**Comment by JakeSidSmith** [comment]
Have made some adjustments.

These mostly fix some styles that were broken in the transition to tailwind, but there are a few others:

- Updated watch script to watch layouts and partials
- Added image to JavaScript Jungle news (was broken link)
- Add "News" to main nav (and remove from footer)
- Add header to "News" section of home page

Screenshot 2024-03-15 at 13 23 57
Screenshot 2024-03-15 at 13 24 10
Screenshot 2024-03-15 at 13 24 21

**Review by JakeSidSmith — APPROVED** [review]

File: _layouts/archive.hbs (hunk 1 above)
Comment (JakeSidSmith) [reply _layouts/archive.hbs]: Removed from `html`, this is now the only class on the `body` and `.wrapper`

**Comment by yanneves** [comment]
Nice improvement in my local environment with these changes, build time in a clean directory improved 5x from 15s -> 3s 🚀

**Pull request merged**